From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: JiaY-shi <shi05275@163.com>
Date: Mon, 31 Aug 2026 02:00:00 +0800
Subject: [PATCH] arm64: dts: mediatek: gl-mt2500: add ramoops

Retain the ramoops area used by the vendor/OpenWrt memory map. The
adjacent BL31 region is already reserved by the upstream MT7981 DTSI.

Signed-off-by: JiaY-shi <shi05275@163.com>

---
 .../dts/mediatek/mt7981b-glinet-gl-mt2500.dtsi | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/arch/arm64/boot/dts/mediatek/mt7981b-glinet-gl-mt2500.dtsi b/arch/arm64/boot/dts/mediatek/mt7981b-glinet-gl-mt2500.dtsi
index 111111111111..222222222222 100644
--- a/arch/arm64/boot/dts/mediatek/mt7981b-glinet-gl-mt2500.dtsi
+++ b/arch/arm64/boot/dts/mediatek/mt7981b-glinet-gl-mt2500.dtsi
@@ -78,6 +78,14 @@ reg_3p3v: regulator-3p3v {
 	};
 };
 
+&{/reserved-memory} {
+	ramoops@42ff0000 {
+		compatible = "ramoops";
+		reg = <0 0x42ff0000 0 0x10000>;
+		record-size = <0x1000>;
+	};
+};
+
 &pio {
 	mmc0_pins_default: mmc0-pins-default {
 		mux {
-- 
2.47.2
